`save_parameters` raises bad error when `dataset` is numpy array

### What happened? (You can include a screenshot if it helps explain)

Calling `save_parameters` with a numpy array raises the following:

```python
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Cell In[30], line 9
      7 experiment = storage.get_experiment_by_name(experiment_name)
      8 posterior_ensemble = experiment.create_ensemble(ensemble_size=ensemble_size, name="poly-posterior", prior_ensemble=ensemble)
----> 9 posterior_ensemble.save_parameters(group="COEFFS", realization=None, dataset=X_posterior_ESMDA)

File [~/git/ert/src/ert/storage/mode.py:98](http://localhost:8889/lab/tree/~/git/ert/src/ert/storage/mode.py#line=97), in require_write.<locals>.inner(self_, *args, **kwargs)
     95 @wraps(func)
     96 def inner(self_: C, [/](http://localhost:8889/), *args: P.args, **kwargs: P.kwargs) -> T:
     97     self_.assert_can_write()
---> 98     return func(self_, *args, **kwargs)

File [~/git/ert/src/ert/storage/local_ensemble.py:829](http://localhost:8889/lab/tree/~/git/ert/src/ert/storage/local_ensemble.py#line=828), in LocalEnsemble.save_parameters(self, group, realization, dataset)
    826     self._storage._to_parquet_transaction(group_path, df_full)
    827     return
--> 829 assert realization is not None, (
    830     "Realization must be provided for xarray Dataset"
    831 )
    832 if "values" not in dataset.variables:
    833     raise ValueError(
    834         f"Dataset for parameter group '{group}' "
    835         "must contain a 'values' variable"
    836     )

AssertionError: Realization must be provided for xarray Dataset
```

### What did you expect to happen?

I expect a better error message such as:

`dataset` must be either an xarray Dataset or polars Dataframe

### Steps to reproduce

```Python

```

### Environment where bug has been observed

- [ ] Python 3.11
- [ ] Python 3.12
- [ ] macOS
- [ ] RHEL8
- [ ] RHEL9
- [ ] Local queue
- [ ] LSF queue
- [ ] Slurm queue
- [ ] OpenPBS queue

--- a/src/ert/storage/local_ensemble.py
+++ b/src/ert/storage/local_ensemble.py
@@ -803,6 +803,11 @@ class LocalEnsemble(BaseMode):
         Saves the provided dataset under a parameter group and realization index(es)
 
         """
+        assert isinstance(dataset, (xr.Dataset | pl.DataFrame)), (
+            f"Dataset must be either an xarray Dataset or polars Dataframe, "
+            f"was '{type(dataset).__name__}'"
+        )
+
         if isinstance(dataset, pl.DataFrame):
             if dataset.is_empty():
                 raise ValueError("Parameters dataframe is empty.")

--- a/tests/ert/unit_tests/storage/test_local_storage.py
+++ b/tests/ert/unit_tests/storage/test_local_storage.py
@@ -21,6 +21,7 @@ from hypothesis import assume, given, note, settings
 from hypothesis.extra.numpy import arrays
 from hypothesis.stateful import Bundle, RuleBasedStateMachine, initialize, rule
 from pandas import DataFrame, ExcelWriter
+from pydantic import BaseModel
 
 from ert.config import (
     DesignMatrix,
@@ -42,8 +43,8 @@ from ert.storage import (
     RealizationStorageState,
     open_storage,
 )
-from ert.storage.local_storage import _LOCAL_STORAGE_VERSION
-from ert.storage.mode import ModeError
+from ert.storage.local_storage import _LOCAL_STORAGE_VERSION, LocalStorage
+from ert.storage.mode import Mode, ModeError
 from tests.ert.unit_tests.config.egrid_generator import egrids
 from tests.ert.unit_tests.config.summary_generator import summaries, summary_variables
 
@@ -123,6 +124,26 @@ def test_that_saving_empty_responses_fails_nicely(tmp_path):
             ),
         ):
             ensemble.save_response("RESPONSE", empty_data, 0)
+
+
+def test_that_local_ensemble_save_parameter_raises_value_error_given_xr_array_dataset(
+    monkeypatch,
+):
+    # Mock storage interactions
+    monkeypatch.setattr(Path, "read_text", lambda *args, **kwargs: None)
+    monkeypatch.setattr(BaseModel, "model_validate_json", lambda _: True)
+    monkeypatch.setattr(LocalStorage, "_save_index", lambda _: None)
+
+    # Create read only storage but pretend it's write to avoid lock files
+    local_storage = LocalStorage(path=Path(""), mode=Mode.READ)
+    local_ensemble = LocalEnsemble(storage=local_storage, path=Path(""), mode=Mode.READ)
+    monkeypatch.setattr(LocalEnsemble, "can_write", lambda _: True)
+
+    expected_error_msg = (
+        r"Dataset must be either an xarray Dataset or polars Dataframe, was 'ndarray'"
+    )
+    with pytest.raises(AssertionError, match=expected_error_msg):
+        local_ensemble.save_parameters(dataset=np.array([]), group="", realization=None)
 
 
 def test_that_saving_response_updates_configs(tmp_path):

